### PR TITLE
STCOM-429: Fixed footer for Save and Cancel buttons

### PR DESCRIPTION
**Pull request by SergiyVSergiyenko**
### Purpose
Implement footer with _Save & close_ and _Canclel_ buttons on the edit user form according to [story](https://issues.folio.org/browse/STCOM-429).

Image

### CODE REVIEW

File: src/UserForm.js [hunk 1]
```diff
@@ -292,25 +278,44 @@ class UserForm extends React.Component {
...
+      pristine,
+      submitting,
+      invalid,
+      onCancel,
+      buttonStyle,
+      cancelButtonStyle,
+    } = this.props;
+
+    const disabled = pristine || submitting || invalid;
+    const testId = id || uniqueId('edituser-');
```
Comment (YevheniiMaltsev) [inline src/UserForm.js]: Could you please clarify why it's named 'testId'?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## 2.25.0 (IN PROGRESS)
 
 * Implement exporting of only overdue loans for overdue loans report. Refs UIU-1121.
+* Create a footer, using <PaneFooter> component, and pass it as an optional prop for <Pane> component. Refs STCOM-429.
```
Comment (VictorSoroka1) [inline CHANGELOG.md]:
```suggestion
* Implement footer with _Save & close_ and _Canclel_ buttons on the edit user form. Refs STCOM-429.
```

Please take out the implementation details and be more specific about the feature.

File: src/UserForm.js [hunk 3]
```diff
@@ -292,25 +278,44 @@ class UserForm extends React.Component {
...
-      >
-        <Icon icon="times-circle">
-          <FormattedMessage id="ui-users.cancel" />
-        </Icon>
-      </Button>
+      <PaneFooter>
+        <Button
+          data-test-user-form-pane-footer-cancel-button
+          buttonStyle={cancelButtonStyle}
+          id={`clickable-${testId}-cancel`}
```
Comment (VictorSoroka1) [inline src/UserForm.js]: I suggest to get rid of `id` attributes and use `data-test` instead.

File: src/UserForm.js [hunk 4]
```diff
@@ -292,25 +278,44 @@ class UserForm extends React.Component {
...
+          onClick={onCancel}
+        >
+          {cancelLabel}
+        </Button>
+        <Button
+          data-test-user-form-pane-footer-submit-button
+          buttonStyle={buttonStyle}
+          id={`clickable-${testId}-submit`}
```
Comment (VictorSoroka1) [inline src/UserForm.js]: I suggest to get rid of `id` attributes and use `data-test` instead.

File: src/UserForm.js [hunk 5]
```diff
@@ -131,14 +132,20 @@ class UserForm extends React.Component {
     parentResources: PropTypes.object.isRequired,
     pristine: PropTypes.bool,
     submitting: PropTypes.bool,
+    buttonStyle: PropTypes.string,
```
Comment (YevheniiMaltsev) [inline src/UserForm.js]: This style is applied to submit button according to the code below. Could you please clarify why the label is named 'submitLabel'  but the put style is named like 'buttonStyle'?

File: src/UserForm.js [hunk 6]
```diff
@@ -292,25 +278,44 @@ class UserForm extends React.Component {
...
+      submitting,
+      invalid,
+      onCancel,
+      buttonStyle,
+      cancelButtonStyle,
+    } = this.props;
+
+    const disabled = pristine || submitting || invalid;
+    const testId = id || uniqueId('edituser-');
+    const cancelLabel = this.props.cancelLabel || <FormattedMessage id="ui-users.cancel" />;
```
Comment (VictorSoroka1) [inline src/UserForm.js]: I suggest removing `this.props.cancelLabel` prop at all as if labels will need to be changed, there will be only change for `FormattedMessage`.

File: src/UserForm.js [hunk 7]
```diff
@@ -292,25 +278,44 @@ class UserForm extends React.Component {
...
+    const submitLabel = this.props.submitLabel || <FormattedMessage id="ui-users.saveAndClose" />;
```
Comment (VictorSoroka1) [inline src/UserForm.js]: I suggest removing `this.props. submitLabel ` prop at all as if labels will need to be changed, there will be only change for `FormattedMessage`.

File: src/UserForm.js [hunk 8]
```diff
@@ -292,25 +278,44 @@ class UserForm extends React.Component {
...
-        buttonStyle="dropdownItem"
-        onClick={handleClick}
-      >
-        <Icon icon="times-circle">
-          <FormattedMessage id="ui-users.cancel" />
-        </Icon>
-      </Button>
+      <PaneFooter>
+        <Button
+          data-test-user-form-pane-footer-cancel-button
```
Comment (VictorSoroka1) [inline src/UserForm.js]:
```suggestion
          data-test-user-form-cancel-button
```

File: src/UserForm.js [hunk 9]
```diff
@@ -292,25 +278,44 @@ class UserForm extends React.Component {
...
+          buttonStyle={cancelButtonStyle}
+          id={`clickable-${testId}-cancel`}
+          onClick={onCancel}
+        >
+          {cancelLabel}
+        </Button>
+        <Button
+          data-test-user-form-pane-footer-submit-button
```
Comment (VictorSoroka1) [inline src/UserForm.js]:
```suggestion
          data-test-user-form-submit-button
```

File: src/UserForm.js [hunk 10]
```diff
@@ -292,25 +275,38 @@ class UserForm extends React.Component {
...
-          <FormattedMessage id="ui-users.cancel" />
-        </Icon>
-      </Button>
+      <PaneFooter>
+        <Button
+          data-test-user-form-cancel-button
+          buttonStyle={cancelButtonStyle}
+          onClick={onCancel}
+        >
+          {<FormattedMessage id="ui-users.cancel" />}
```
Comment (VictorSoroka1) [inline src/UserForm.js]:
```suggestion
         <FormattedMessage id="ui-users.cancel" />
```

File: src/UserForm.js [hunk 11]
```diff
@@ -292,25 +275,38 @@ class UserForm extends React.Component {
...
+        </Button>
+        <Button
+          data-test-user-form-submit-button
+          buttonStyle={submitButtonStyle}
+          type="submit"
+          disabled={disabled}
+        >
+          {<FormattedMessage id="ui-users.saveAndClose" />}
```
Comment (VictorSoroka1) [inline src/UserForm.js]:
```suggestion
          <FormattedMessage id="ui-users.saveAndClose" />
```

File: src/UserForm.js [hunk 12]
```diff
@@ -292,25 +275,38 @@ class UserForm extends React.Component {
...
+        <Button
+          data-test-user-form-cancel-button
+          buttonStyle={cancelButtonStyle}
+          onClick={onCancel}
+        >
+          {<FormattedMessage id="ui-users.cancel" />}
+        </Button>
+        <Button
+          data-test-user-form-submit-button
+          buttonStyle={submitButtonStyle}
```
Comment (VictorSoroka1) [inline src/UserForm.js]: I suggest removing `submitButtonStyle` and `cancelButtonStyle` properties at all and use values which you have specified in default props for them.

**Pull request merged**